cpu/numactl.py

```diff
@@ -67,6 +74,28 @@ def setUp(self):
 
         build.make(self.sourcedir)
 
+        interfaces = netifaces.interfaces()
+        interface = self.params.get("interface")
+        if interface not in interfaces:
+            self.cancel("%s interface is not available" % interface)
+        self.iface = interface
+        local = LocalHost()
+        self.networkinterface = NetworkInterface(self.iface, local)
+        self.ipaddr = self.params.get("host_ip", default="")
+        self.netmask = self.params.get("netmask", default="")
+        self.peer = self.params.get("peer_ip")
+        if not self.peer:
+            self.cancel("No peer provided")
+        if self.networkinterface.ping_check(self.peer, count=5) is not None:
```

Comment:
@FarooqAbdulla01 I think you need to check if the interfaces are up or not before checking ping.